docs/proposals/scheduling/20241008-gpu-partition-api.md

```diff
@@ -0,0 +1,225 @@
+
+### Goals
+- Provide the API for a Pod to request a specific GPU Partition.
+- Allow nodes to offer permitted Partitions and whether Pods are guaranteed to be allocated based on the partition.
+
+### Non-Goals/Future Work
+- Describe what the Partition Table looks like for a specific GPU model.
+
+## User Story
+
+Typically, the rules for GPU partitioning are determined by the specific GPU model or system configuration, and may also be influenced by the configuration of GPUs on each individual node. The scheduler does not have insight into the specifics of the hardware models or GPU types; instead, it relies on components at the node level to report these Partition Rules to the Device Custom Resource (CR) as follows:
+
+```yaml
+apiVersion: scheduling.koordinator.sh/v1alpha1
+kind: Device
+metadata:
+  annotations:
+    scheduling.koordinator.sh/gpu-partitions: |
+      {
+        "1": [
+            "NVLINK": {
+                {
+                  # Which GPUs are included
+                  "Minors": [
+                      0
+                  ],
+                  # GPU Interconnect Type
+                  "gpuLinkType": "NVLink",
+                  # GPU Interconnect Bandwidth
+                  "gpuLinkBandWidth": 400Gi
+                },
+                ...
+            }
+            ...
+        ],
+        "2": [
+            ...
+        ],
+        "4": [
+            ...
+        ],
+        "8": [
+            ...
+        ]
+      }
+  labels:
+    node.koordinator.sh/gpu-partition-allocate-policy: "Honor"
+  name: node-1
+```
+
+Users to specify the desired GPU partitioning configuration at the Pod level.
+
+```yaml
+kind: Pod
+metadata:
+  name: hello-gpu
+  annotations:
+    scheduling.koordinator.sh/gpu-partition-spec: |
+      {
+        "allocatePolicy": "Prefer",
+        "gpuLinkType": "NVLink",
+        "gpuLinkBandwidth": "200Gi"
+      }
+spec:
+  containers:
+    - name: main
+      resources:
+        limits:
+          nvidia.com/gpu: 1
+```
+
+## Proposal
+
+### GPUPartitionTable
+
+Before we proceed, let's define the term "Partition" step by step. A Partition refers to a combination of GPUs that can be allocated to a user, possessing the following attributes:
+
+```yaml
+{
+    # Which GPUs are included
+    "Minors": [
+        0
+    ],
+    # GPU Interconnect Type
+    "gpuLinkType": "NVLink",
+    # GPU Interconnect Bandwidth
+    "gpuLinkBandWidth": 400Gi
+},
+```
+
+Combining all possible combinations of partitions yields a partition table.
+
+```yaml
+{
+    "1": [
+        "NVLINK": {
+        	{
+              # Which GPUs are included
+              "Minors": [
+                  0
+              ],
+              # GPU Interconnect Type
+              "gpuLinkType": "NVLink",
+              # GPU Interconnect Bandwidth
+              "gpuLinkBandWidth": 400Gi
+        	},
+        	...
+        }
+        ...
+    ],
+    "2": [
```

Comment:
Suggestion: to be more clear, we can explain here with "The key here in a partition table is an integer that reflects how many GPU cards are included in this partition group. " or something.